### PR TITLE
Add feed

### PR DESCRIPTION
**Pull request by hisbaan**
This branch adds feed-adding functionality. When the user clicks the add feed button (starting the AddFeedActivity), they will be presented with a text field in which to enter a link. Upon submitting the field, the feed, if valid, will be added to the main feed group.

### CODE REVIEW

File: app/src/main/java/monster/minions/binocularss/dataclasses/Feed.kt [hunk 1]
```diff
@@ -26,15 +26,46 @@ import monster.minions.binocularss.room.TagsListConverter
...
     var articles: MutableList<Article> = mutableListOf(),
     var tags: MutableList<String> = mutableListOf(),
     var priority: Int = 0,
-) : Parcelable
\ No newline at end of file
+) : Parcelable {
+    /**
+     * Check if a feed is equal to another by checking the source, which usually does not change
+     *
+     * @param other Another object. If it is not an article, return false immediately.
```
Comment (ismail-ahmed0149) [inline app/src/main/java/monster/minions/binocularss/dataclasses/Feed.kt:45]: Should be "If it is not a Feed"

Comment (hisbaan) [reply app/src/main/java/monster/minions/binocularss/dataclasses/Feed.kt:45]: Will fix. Thanks

**Review by ismail-ahmed0149 — APPROVED** [review]
Everything seems to be looking good!

**Review by ismail-ahmed0149 — APPROVED** [review]

**Review by Just-Call-Sal — APPROVED** [review]
I really like the idea of adding some feeds to the user's feedGroup if their feedGroup is empty, we can extend this to significantly improve the user experience.  I approve merging these changes into the main branch.

**Pull request merged**